store/postgres/src/relational.rs

```diff
@@ -541,14 +582,22 @@ impl Column {
         is_list(&self.field_type)
     }
 
+    pub fn is_enum(&self) -> bool {
+        self.enum_type.is_some()
+    }
+
     /// Generate the DDL for one column, i.e. the part of a `create table`
     /// statement for this column.
     ///
     /// See the unit tests at the end of this file for the actual DDL that
     /// gets generated
     fn as_ddl(&self, out: &mut String) -> fmt::Result {
         write!(out, "    ")?;
-        write!(out, "{:20} {}", self.name, self.sql_type())?;
+        let type_name = match &self.enum_type {
+            Some(ref type_name) => type_name.as_str(),
+            None => self.sql_type(),
+        };
```

Comment:
This code might indicate the abstraction is not quite right, I'd expect `Column::sql_type` to return the enum type since enums are also sql. Maybe enum types should be a variant of `ColumnType`?

Reply:
True; I tried it with a `ColumnType::Enum(_)` variant, and that does clean up the code quite a bit.

Reply:
Nice!